src/index.js

```diff
@@ -36,16 +73,51 @@ const extendCodeBlockWithSyntaxHighlighting = ( settings ) => {
 			language: {
 				type: 'string',
 			},
+			selectedLines: {
+				type: 'string',
+			},
 			showLines: {
 				type: 'boolean',
 			},
 		},
 
 		edit( { attributes, setAttributes, className } ) {
+			const parseSelectedLines = ( selectedLines ) => {
+				const highlightedLines = new Set();
+
+				if ( ! selectedLines || selectedLines.trim().length === 0 ) {
+					return highlightedLines;
+				}
+
+				let chunk;
+				const ranges = selectedLines.replace( /\s/, '' ).split( ',' );
+
+				for ( chunk of ranges ) {
+					if ( chunk.indexOf( '-' ) >= 0 ) {
+						let i;
+						const range = chunk.split( '-' );
+
+						if ( range.length === 2 ) {
+							for ( i = +range[ 0 ]; i <= +range[ 1 ]; ++i ) {
+								highlightedLines.add( i - 1 );
+							}
+						}
+					} else {
+						highlightedLines.add( +chunk - 1 );
+					}
+				}
+
+				return highlightedLines;
+			};
```

Comment:
Minor thing: how about moving `parseSelectedLines` into a separate file or at least out of the `edit` function here? That would avoid it getting re-created each time the `edit` function runs.

Reply:
Ah good catch! I hadn't noticed this was inside of `edit`. Moved it outside of edit